components/script/dom/gpudevice.rs

```diff
@@ -478,7 +497,12 @@ impl GPUDeviceMethods for GPUDevice {
         descriptor: &GPURenderPipelineDescriptor,
     ) -> Fallible<DomRoot<GPURenderPipeline>> {
         let (implicit_ids, desc) = self.parse_render_pipeline(descriptor)?;
-        let render_pipeline = GPURenderPipeline::create(self, implicit_ids, desc, None)?;
+        let render_pipeline = GPURenderPipeline::create(
+            self,
+            PipelineLayout::Implicit(implicit_ids.implicit().unwrap()),
```

Comment:
The unwrap here will panic whenever the `implicit_id` is `Explicit` whereas previously we simply just pass `None` to `create`. I think we just need to pass in the `pipeline_layout` from `parse_render_pipeline`.